### DISCUSSION TITLE
Since 2.0.7 can't build on centos 6 and centos 7

### DISCUSSION THREAD

**Discussion by gexgik** [fluent/fluent-bit#6872] [General]
Hello, 

Since version 2.0.7 we can no longer build on centos 6 and 7. It seems the reason is the missing folder:
lib/cfl/lib/xxhash/cli .

Snippet:
CMake Error at lib/cfl/lib/xxhash/cmake_unofficial/CMakeLists.txt:91 (add_executable):
Cannot find source file:
     /app/fluent-bit-2.0.8/lib/cfl/lib/xxhash/cli/xxhsum.c
Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
   .hpp .hxx .in .txx

CMake Error at lib/cfl/lib/xxhash/cmake_unofficial/CMakeLists.txt:91 (add_executable):

   No SOURCES given to target: xxhsum

Can you please reinclude the cli folder for xxhash?

**Comment by patrick-stephens** [comment]
It probably needs a bit more detail on how your build machine is set up.

The official builds just checkout the repo and build it for centos/7 fine (also for centos/6 using the same approach for some Enterprise packages I maintain):

```shell
git checkout https://github.com/fluent/fluent-bit.git
./fluent-bit/packaging/build.sh -d centos/7
```

This uses a container to build each target (to provide a clean build from a base image plus allow us to run on generic ephemeral Github runners).

What do you do? 
Any chance you can provide a reproducible example, ideally with a container to make it simple to pick up plus start with a clean image?

We made some changes on the dependencies a while back to properly support Golang plugin builds, plus the dependencies themselves are also versioned/updated independently. What versions (if any) of cmetrics and cfl are you installing locally?

**Reply by ewingate00** [reply]
@patrick-stephens If I run those commands you mentioned for centos 6, it fails with the following errors.
`./fluent-bit/packaging/build.sh -d centos/6`

```
ERROR: failed to solve: centos-6-base: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed (did you mean centos-7-base?)
+ echo 'Error building main docker image flb-centos/6'
Error building main docker image flb-centos/6
+ exit 1
```

**Reply by patrick-stephens** [reply]
Yes, CentOS 6 is not supported by OSS. 

I merely mentioned we do support it for Enterprise offerings using the same process as the supported builds. Ignore it, it's just confusing things!
